src/error/rust.rs

```diff
@@ -142,10 +147,31 @@ where
     }
 }
 
+/// If `result` is Ok, this returns the Ok value and clears [errno].
+/// Otherwise it returns the default value, writes the error message to `errout` and sets [errno].
+///
+/// [errno]: https://utcc.utoronto.ca/~cks/space/blog/programming/GoCgoErrorReturns
+pub fn handle_c_error_default<T>(result: Result<T, RustError>, errout: Option<&mut Buffer>) -> T
```

Comment:
It seems a good choice to split into two functions, remove some abstraction but make both cases clearer